lib/Runtime/Library/AtomicsObject.cpp

```diff
@@ -0,0 +1,270 @@
+    {
+        ATOMICS_FUNCTION_BEGIN_WRAPPER(1, "Atomics.isLockFree");
+        uint32 size = JavascriptConversion::ToUInt32(args[1], scriptContext);
+        return (size == 1 || size == 2 || size == 4) ? scriptContext->GetLibrary()->GetTrue() : scriptContext->GetLibrary()->GetFalse();
+    }
+
+    Var AtomicsObject::EntryLoad(RecyclableObject* function, CallInfo callInfo, ...)
+    {
+        ATOMICS_FUNCTION_BEGIN_WRAPPER(2, "Atomics.load");
+
+        uint32 accessIndex = 0;
+        TypedArrayBase *typedArrayBase = ValidateAndGetTypedArray(args[1], args[2], &accessIndex, scriptContext);
+        return typedArrayBase->TypedLoad(accessIndex);
+    }
+
+    Var AtomicsObject::EntryOr(RecyclableObject* function, CallInfo callInfo, ...)
+    {
+        ATOMICS_FUNCTION_BEGIN_WRAPPER(3, "Atomics.or");
+
+        uint32 accessIndex = 0;
+        TypedArrayBase *typedArrayBase = ValidateAndGetTypedArray(args[1], args[2], &accessIndex, scriptContext);
+        return typedArrayBase->TypedOr(accessIndex, args[3]);
+    }
+
+    Var AtomicsObject::EntryStore(RecyclableObject* function, CallInfo callInfo, ...)
+    {
+        ATOMICS_FUNCTION_BEGIN_WRAPPER(3, "Atomics.store");
+
+        uint32 accessIndex = 0;
+        TypedArrayBase *typedArrayBase = ValidateAndGetTypedArray(args[1], args[2], &accessIndex, scriptContext);
+        return typedArrayBase->TypedStore(accessIndex, args[3]);
+    }
+
+    Var AtomicsObject::EntrySub(RecyclableObject* function, CallInfo callInfo, ...)
+    {
+        ATOMICS_FUNCTION_BEGIN_WRAPPER(3, "Atomics.sub");
+
+        uint32 accessIndex = 0;
+        TypedArrayBase *typedArrayBase = ValidateAndGetTypedArray(args[1], args[2], &accessIndex, scriptContext);
+        return typedArrayBase->TypedSub(accessIndex, args[3]);
+    }
+
+    Var AtomicsObject::EntryWait(RecyclableObject* function, CallInfo callInfo, ...)
+    {
+        ATOMICS_FUNCTION_BEGIN_WRAPPER(3, "Atomics.wait");
+
+        uint32 accessIndex = 0;
+        TypedArrayBase *typedArrayBase = ValidateAndGetTypedArray(args[1], args[2], &accessIndex, scriptContext, true /*onlyInt32*/);
+
+        Var value = args[3];
+        Var timeoutVar = args[4];
+        uint32 timeout = INFINITE;
+
+        if (args.Info.Count > 4 && !JavascriptOperators::IsUndefinedObject(timeoutVar))
+        {
+            double t =JavascriptConversion::ToNumber(timeoutVar, scriptContext);
+            if (!(NumberUtilities::IsNan(t) || JavascriptNumber::IsPosInf(t)))
+            {
+                int32 t1 = JavascriptConversion::ToInt32(t);
+                timeout = (uint32)max(0, t1);
+            }
+        }
+
+        if (!AgentOfBuffer::AgentCanSuspend(scriptContext))
+        {
+            JavascriptError::ThrowTypeError(scriptContext, JSERR_CannotSuspendBuffer);
+        }
+
+        Assert(typedArrayBase->GetBytesPerElement() == 4);
+        uint32 bufferIndex = (accessIndex * 4) + typedArrayBase->GetByteOffset();
```

Comment:
Arraybuffer.transfer can work on the ArrayBuffer only. It is a typeerror for other objects.